signing/multisig/blsMultisig_test.go

```diff
@@ -290,232 +333,447 @@ func TestBLSMultiSigner_CreateOK(t *testing.T) {
+
+		assert.Nil(t, err)
+		assert.Equal(t, sigShare, sigShareRead)
+	})
 }
 
 func TestBLSMultiSigner_SignatureShareOutOfBoundsIndexShouldErr(t *testing.T) {
 	t.Parallel()
 
 	ownIndex := uint16(3)
 	hasher := &mock.HasherSpongeMock{}
-	privKey, _, pubKeys, kg, _ := generateMultiSigParamsBLS(4, ownIndex)
+	llSigner := &llsig.BlsMultiSigner{Hasher: hasher}
+	llSignerKOSK := &llsig.BlsMultiSignerKOSK{}
+	privKey, _, pubKeys, kg := generateMultiSigParamsBLS(4, ownIndex)
 	msg := []byte("message")
-	sigShare, multiSig := createSignerAndSigShareBLS(hasher, pubKeys, privKey, kg, ownIndex, msg)
-	_ = multiSig.StoreSignatureShare(ownIndex, sigShare)
-	sigShareRead, err := multiSig.SignatureShare(15)
 
-	assert.Nil(t, sigShareRead)
-	assert.Equal(t, crypto.ErrIndexOutOfBounds, err)
+	t.Run("with rogue key prevention", func(t *testing.T) {
+		sigShare, multiSig := createSignerAndSigShareBLS(pubKeys, privKey, kg, ownIndex, msg, llSigner)
+		_ = multiSig.StoreSignatureShare(ownIndex, sigShare)
+		sigShareRead, err := multiSig.SignatureShare(15)
+
+		assert.Nil(t, sigShareRead)
+		assert.Equal(t, crypto.ErrIndexOutOfBounds, err)
+	})
+	t.Run("with KOSK", func(t *testing.T) {
+		sigShare, multiSig := createSignerAndSigShareBLS(pubKeys, privKey, kg, ownIndex, msg, llSignerKOSK)
+		_ = multiSig.StoreSignatureShare(ownIndex, sigShare)
+		sigShareRead, err := multiSig.SignatureShare(15)
+
+		assert.Nil(t, sigShareRead)
+		assert.Equal(t, crypto.ErrIndexOutOfBounds, err)
+	})
 }
 
 func TestBLSMultiSigner_SignatureShareNotSetIndexShouldErr(t *testing.T) {
 	t.Parallel()
 
 	ownIndex := uint16(3)
 	hasher := &mock.HasherSpongeMock{}
-	privKey, _, pubKeys, kg, _ := generateMultiSigParamsBLS(4, ownIndex)
+	llSigner := &llsig.BlsMultiSigner{Hasher: hasher}
+	llSignerKOSK := &llsig.BlsMultiSignerKOSK{}
+	privKey, _, pubKeys, kg := generateMultiSigParamsBLS(4, ownIndex)
 	msg := []byte("message")
-	sigShare, multiSig := createSignerAndSigShareBLS(hasher, pubKeys, privKey, kg, ownIndex, msg)
-	_ = multiSig.StoreSignatureShare(ownIndex, sigShare)
-	sigShareRead, err := multiSig.SignatureShare(2)
 
-	assert.Nil(t, sigShareRead)
-	assert.Equal(t, crypto.ErrNilElement, err)
+	t.Run("with rogue key prevention", func(t *testing.T) {
+		sigShare, multiSig := createSignerAndSigShareBLS(pubKeys, privKey, kg, ownIndex, msg, llSigner)
+		_ = multiSig.StoreSignatureShare(ownIndex, sigShare)
+		sigShareRead, err := multiSig.SignatureShare(2)
+
+		assert.Nil(t, sigShareRead)
+		assert.Equal(t, crypto.ErrNilElement, err)
+	})
+	t.Run("with KOSK", func(t *testing.T) {
+		sigShare, multiSig := createSignerAndSigShareBLS(pubKeys, privKey, kg, ownIndex, msg, llSignerKOSK)
+		_ = multiSig.StoreSignatureShare(ownIndex, sigShare)
+		sigShareRead, err := multiSig.SignatureShare(2)
+
+		assert.Nil(t, sigShareRead)
+		assert.Equal(t, crypto.ErrNilElement, err)
+	})
 }
 
 func TestBLSMultiSigner_SignatureShareOK(t *testing.T) {
 	t.Parallel()
 
```

Comment:
`AddSignatureShareOK` from above looks very similar with this test, maybe a duplicate?

Reply:
right, removed.